coalib/bearlib/abstractions/ExternalBearWrap.py

```diff
@@ -0,0 +1,200 @@
+            for key, val in opt_params.items():
+                if key not in settings:
+                    settings[key] = val[2]
+
+        @staticmethod
+        def get_severity(result):
+            """
+            Returns the severity corresponding to the one from the
+                parsed JSON
+
+            :param result: A result dict extracted from the parsed JSON
+
+            :return: A RESULT_SEVERITY
+            """
+            severity_map = {'MAJOR': RESULT_SEVERITY.MAJOR,
+                            'NORMAL': RESULT_SEVERITY.NORMAL,
+                            'INFO': RESULT_SEVERITY.INFO}
+
+            if not 'severity' in result:
+                return RESULT_SEVERITY.NORMAL
+            else:
+                sev = result['severity']
+                return severity_map[sev]
+
+        def parse_output(self, out, filename):
+            """
+            Parses the output JSON into Result objects
+
+            :param out: Raw output from the given executable (should be JSON)
+            :param filename: The filename of the analyzed file. Needed to
+                create the Result objects.
+
+            :yield: The Result objects.
+            """
+            output = json.loads(out)
+
+            for result in output:
+                yield Result.from_values(
+                    origin=self,
+                    message=result['message'],
+                    file=filename,
+                    line=result['line'],
+                    end_line=result.get('end_line', None),
+                    column=result.get('column', None),
+                    end_column=result.get('end_column', None),
+                    severity=self.get_severity(result),
+                    debug_msg=result.get('debug_msg', ""),
+                    additional_info=result.get('additional_info', ""))
+
+        def run(self, filename, file, **settings):
+            self._prepare_settings(settings)
+            json_string = json.dumps({'filename': filename,
+                                      'file': file,
+                                      'settings': settings})
+
+            args = self.create_arguments()
+            try:
+                args = tuple(args)
+            except TypeError:
+                self.err("The given arguments "
+                         "{!r} are not iterable.".format(args))
+                return
+
+            shell_command = (self.get_executable(),) + args
+            out, err = run_shell_command(shell_command, json_string)
+
+            return self.parse_output(out, filename)
+
+    result_klass = type(klass.__name__, (klass, ExternalBearWrapBase), {})
+    result_klass.__doc__ = klass.__doc__ if klass.__doc__ else ""
+    return result_klass
+
+
+@enforce_signature
+def external_bear_wrap(executable: str, **options):
+
+    options["executable"] = executable
+    _prepare_options(options)
+
+    return partial(_create_wrapper, options=options)
```

Comment:
one thing I would improve for this file is the general documentation style, either use the "old" or the "new" one:
```
# old style:
:param x:    Hello world x
:param xyzw: xyzw

#new style
:param x:
    x-desc
:param xyzw:
    xyzw
```